delivery_total_weight_from_packaging/models/stock_quant_package.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 Camptocamp SA
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
+
+from odoo import api, models
+
+
+class StockQuantPackage(models.Model):
+    _inherit = "stock.quant.package"
+
+    @api.depends("quant_ids")
+    @api.depends_context("picking_id")
+    def _compute_weight(self):
+        # Override method from `delivery` module to compute a more accurate
+        # weight by including the weight of the packaging
+        # NOTE: code copied/pasted and adapter from `delivery`
+        for package in self:
+            weight = 0.0
+            if self.env.context.get("picking_id"):
+                current_picking_move_line_ids = self.env["stock.move.line"].search(
```

Comment:
could we avoid a search for each item?

Reply:
Don't know which is the best: querying all move lines for all packages at once, then group them by package, or searching them as it is done right now.

Reply:
Before I update this one, your review is welcome on https://github.com/OCA/stock-logistics-workflow/pull/777